user/views.py

```diff
@@ -206,3 +208,28 @@ def update_password(user, validated_data):
                 data={"message": "Old password is incorrect.", "code": 400},
                 status=status.HTTP_400_BAD_REQUEST,
             )
+
+
+class UserRoadmapDetailAPIView(RetrieveAPIView):
+    queryset = UserRoadmap.objects.all()
+    serializer_class = UserRoadmapDetailSerializer
+    authentication_classes = [CustomJWTAuthentication,]
+    lookup_field = "user"
+    lookup_url_kwarg = "user_pk"
+
+    def get_queryset(self):
+        queryset = self.queryset
```

Comment:
use `self.get_queryset().filter(user=user, roadmap_id=roadmap_pk)` instead of `self.queryset`